src/utils/semaphore.cpp

```diff
@@ -0,0 +1,83 @@
+#include <pangolin/utils/semaphore.h>
+
+#include <fcntl.h>
+#include <semaphore.h>
+#include <sys/stat.h>
```

Comment:
Looks like this isn't going to build on Windows. Not sure about OSX. We will need to conditionally add this stuff depending on platform.